spec/system/admin/reports/revenues_by_hub_spec.rb

```diff
@@ -10,22 +10,67 @@
       :completed_order_with_totals,
       completed_at: 2.days.ago,
       order_cycle:,
-      distributor: create(:enterprise, name: "Hub 1",
-                                       owner: create(:user, email: "email@email.com")),
+      distributor: distributor1
     )
   end
+  let(:order_with_voucher_tax_included) do
+    create(
+      :order_with_taxes,
+      completed_at: 2.days.ago,
+      order_cycle:,
+      distributor: distributor2,
+      product_price: 110.0,
+      tax_rate_amount: 0.1,
+      included_in_price: true,
+      tax_rate_name: "Tax 1"
+    )
+  end
+  let(:order_with_voucher_tax_excluded) do
+    create(
+      :order_with_taxes,
+      completed_at: 2.days.ago,
+      order_cycle:,
+      distributor: distributor3,
+      product_price: 110.0,
+      tax_rate_amount: 0.1,
+      included_in_price: true,
+      tax_rate_name: "Tax 1"
+    )
+  end
+  let(:distributor1) { create(:enterprise, name: "Hub 1", owner:) }
+  let(:distributor2) { create(:enterprise, name: "Hub 2", owner:) }
+  let(:distributor3) { create(:enterprise, name: "Hub 3", owner:) }
+  let(:owner) { create(:user, email: 'email@email.com') }
   let(:order_cycle) { create(:simple_order_cycle) }
   let(:product) { create(:product, supplier:) }
   let(:supplier) { create(:supplier_enterprise) }
 
   before do
     create(:line_item_with_shipment, order:, product:)
 
+    order_with_voucher_tax_included.create_tax_charge!
+    order_with_voucher_tax_included.update_shipping_fees!
+    order_with_voucher_tax_included.update_order!
+
+    order_with_voucher_tax_excluded.create_tax_charge!
+    order_with_voucher_tax_excluded.update_shipping_fees!
+    order_with_voucher_tax_excluded.update_order!
+
+    allow(VoucherAdjustmentsService).to receive(:new) do |order_arg|
+      if order_arg.id == order.id
+        next double(voucher_included_tax: 0.0, voucher_excluded_tax: 0.0)
+      elsif order_arg.id == order_with_voucher_tax_included.id
+        next double(voucher_included_tax: 0.5, voucher_excluded_tax: 0.0)
+      elsif order_arg.id == order_with_voucher_tax_excluded.id
+        next double(voucher_included_tax: 0.0, voucher_excluded_tax: -0.5)
+      end
+    end
```

Comment:
This is too much mocking. We should set up real data.

Reply:
Especially in system spec, we want to test scenarios as close as possible to what a user would experience.

You can have a look here to see how to set up data and apply a voucher to an order : https://github.com/openfoodfoundation/openfoodnetwork/blob/f6ba2dc70c0063d93f907ce4948de20af4b120a2/spec/system/consumer/checkout/summary_spec.rb#L410 
It's a bit tricky to put everything in place, let us know if you need some help.

Reply:
ye okay, let me take a look. I'm a bit busy in weekdays so I'm gonna update it in weekends